<a href="https://colab.research.google.com/github/khalilhaddad7/Titanic-Survival-Analysis-Data-Analysis-Project-/blob/main/Titanic%20Survival%20Analysis%20(Data%20Analysis%20Project).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



<H1>TP3 Analyse de données<H1>

## Contexte
    
Dans ce TP, nous utiliserons l'exemple du jeu de données du Titanic, un énorme paquebot pour l'époque qui fait naufrage en 1912 à la suite d'une collision avec un iceberg, lors de son voyage inaugural de Southampton à New York.
   * Nous avons accès à des informations sur une partie des passagers (1309 passagers) du Titanic.
   * Pourquoi certains passagers ont survécu et d'autres sont morts?
   * Commencons l'analyse de données



## Chargement des données
Les données sont dans un tableau au format CSV (comma separated values):

In [ ]:
import pandas as pd


In [ ]:
df = pd.read_csv("titanic (2).csv")
df

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0.0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1.0,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1.0,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1.0,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0.0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
...,...,...,...,...,...,...,...,...,...,...,...,...
1304,1305,NaN,3,"Spector, Mr. Woolf",male,NaN,0,0,A.5. 3236,8.0500,NaN,S
1305,1306,NaN,1,"Oliva y Ocana, Dona. Fermina",female,39.0,0,0,PC 17758,108.9000,C105,C
1306,1307,NaN,3,"Saether, Mr. Simon Sivertsen",male,38.5,0,0,SOTON/O.Q. 3101262,7.2500,NaN,S
1307,1308,NaN,3,"Ware, Mr. Frederick",male,NaN,0,0,359309,8.0500,NaN,S


Ensuite, on OBSERVE ce qu’on a!!

In [ ]:
df['Adult'] = df['Age'].apply(lambda x: True if x >= 18 else False).fillna(False)
df

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked,Adult
0,1,0.0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S,True
1,2,1.0,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C,True
2,3,1.0,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S,True
3,4,1.0,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S,True
4,5,0.0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...
1304,1305,NaN,3,"Spector, Mr. Woolf",male,NaN,0,0,A.5. 3236,8.0500,NaN,S,False
1305,1306,NaN,1,"Oliva y Ocana, Dona. Fermina",female,39.0,0,0,PC 17758,108.9000,C105,C,True
1306,1307,NaN,3,"Saether, Mr. Simon Sivertsen",male,38.5,0,0,SOTON/O.Q. 3101262,7.2500,NaN,S,True
1307,1308,NaN,3,"Ware, Mr. Frederick",male,NaN,0,0,359309,8.0500,NaN,S,False


In [ ]:
df.shape

(1309, 13)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 13 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  1309 non-null   int64  
 1   Survived     891 non-null    float64
 2   Pclass       1309 non-null   int64  
 3   Name         1309 non-null   object 
 4   Sex          1309 non-null   object 
 5   Age          1046 non-null   float64
 6   SibSp        1309 non-null   int64  
 7   Parch        1309 non-null   int64  
 8   Ticket       1309 non-null   object 
 9   Fare         1308 non-null   float64
 10  Cabin        295 non-null    object 
 11  Embarked     1307 non-null   object 
 12  Adult        1309 non-null   bool   
dtypes: bool(1), float64(3), int64(4), object(5)
memory usage: 124.1+ KB


Les colonnes sont:

* PassengerId : Passenger Id

* Survived : True (1) / False (0)

* Pclass : Passenger ticket class : Class 1, 2 and 3.

* Name : Name of the passenger

* Sex : Sex of the passenger male/female

* Age : Age in years

* Sibsp :	# of siblings / spouses aboard the Titanic

* Parch	: # of parents / children aboard the Titanic

* Ticket :	Ticket number

* fare :	Passenger fare

* Cabin : Cabin number

* Embarked : Embarkation port ( C : Cherbourg; Q Queenstown; S Southampton)

## Question / Objectif
<strong>Pourquoi certains passagers ont survécu et d’autres sont morts?<strong>

On veut trouver les colonnes qui expliquent le 0/1 dans Survived et construire un modèle qui permettrait d’expliquer au mieux Survived étant donné nos informations.

On va commencer par observer nos données, en répondant à des questions descriptives:
1. Quel sexe a le plus de chances de survie ?
2. Est-ce que les enfants ont eu plus de chances de survie ?
3. Calculer la proportion de survie selon le port d'embarquement.

### Etape 1: Supprimer les NAN de la colonnes "Survived"
La colonne "Survived" a des valeurs manquantes (891 disponibles/1309). Comme la plupart des passagers et passagères sont mort, on peut supposer que Survived =0.

Attention ceci est un choix. Toujours garder en tête qu’il modifie vos résultats et peut donc modifier vos interpétations ! Ici, c’est vraiment à la marge

In [ ]:
df['Survived'] = df['Survived'].fillna(0)


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 13 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  1309 non-null   int64  
 1   Survived     1309 non-null   float64
 2   Pclass       1309 non-null   int64  
 3   Name         1309 non-null   object 
 4   Sex          1309 non-null   object 
 5   Age          1046 non-null   float64
 6   SibSp        1309 non-null   int64  
 7   Parch        1309 non-null   int64  
 8   Ticket       1309 non-null   object 
 9   Fare         1308 non-null   float64
 10  Cabin        295 non-null    object 
 11  Embarked     1307 non-null   object 
 12  Adult        1309 non-null   bool   
dtypes: bool(1), float64(3), int64(4), object(5)
memory usage: 124.1+ KB


### Quel sexe a le plus de chances de survie ?
* Comparaison de la proportion d’hommes et de femmes passagers du Titanic qui ont survécu

* Utilisons groupby, qui permet de produire des tables de synthèses par catégories:

In [ ]:
df.groupby(['Sex', 'Survived'])['PassengerId'].count()


Sex     Survived
female  0.0         233
        1.0         233
male    0.0         734
        1.0         109
Name: PassengerId, dtype: int64

#### Autres tables de synthèse

In [ ]:
df['Sex'].value_counts()


,count
Sex,
male,843
female,466


In [ ]:
summary = df.groupby('Sex').agg(
    Surv=('Survived', 'sum'),
    Pass=('PassengerId', 'count')
)
summary['%'] = (summary['Surv'] / summary['Pass'] * 100).round(1)
print(summary)


         Surv  Pass     %
Sex                      
female  233.0   466  50.0
male    109.0   843  12.9


### Est-ce que les enfants ont eu plus de chances de survie ?
* On va commencer par séparer les enfants des adultes selon l’age. Problème, on a des données manquantes.

#### Gestion des données manquantes
Il manque certaines informations. Que feriez-vous ?

On pourrait décider de supprimer les individus sans informations sur l’âge (pensez à vérifier les dimensions de votre table!)

In [ ]:
df_drop_all = df.dropna()
df_drop_age = df.dropna(subset=['Age'])


Maintenant on va créer une nouvelle colonne indiquant si l’on est adulte

In [ ]:
df['Adult'] = df['Age'].apply(lambda x: True if x >= 18 else False)

Et qui des individus dont on ne connait pas l’age ?

In [ ]:
df.groupby(['Adult', 'Sex'])['PassengerId'].count()


Adult  Sex   
False  female     72
       male       82
True   female    394
       male      761
Name: PassengerId, dtype: int64

In [ ]:
df.groupby(['Adult', 'Sex'])['Survived'].sum()


Adult  Sex   
False  female     38.0
       male       23.0
True   female    195.0
       male       86.0
Name: Survived, dtype: float64

In [ ]:
def category(row):
    if not row['Adult'] and row['Sex'] == 'female':
        return 'Girl'
    elif not row['Adult'] and row['Sex'] == 'male':
        return 'Boy'
    elif row['Adult'] and row['Sex'] == 'female':
        return 'Woman'
    else:
        return 'Man'

df['Category'] = df.apply(category, axis=1)


summary_cat = df.groupby('Category').agg(
    Survivants=('Survived', 'sum'),
    Passagers=('PassengerId', 'count')
)
summary_cat['%'] = (summary_cat['Survivants'] / summary_cat['Passagers'] * 100).round(1)

print(summary_cat)


          Survivants  Passagers     %
Category                             
Boy             23.0         82  28.0
Girl            38.0         72  52.8
Man             86.0        761  11.3
Woman          195.0        394  49.5


### Calculer la proportion de survie selon le port d’embarquement

La colonne du port d’embarquement a des valeurs manquantes. Comme la plupart des passagers et passagères sont montées à Southampton, on peut supposer que les données manquantes viennent de là.


In [ ]:
summary_embarked = df.groupby('Embarked').agg(
    Survivants=('Survived', 'sum'),
    Passagers=('PassengerId', 'count')
)
summary_embarked['%'] = (summary_embarked['Survivants'] / summary_embarked['Passagers'] * 100).round(1)

print(summary_embarked)


          Survivants  Passagers     %
Embarked                             
C               93.0        270  34.4
Q               30.0        123  24.4
S              217.0        914  23.7


Retour au port d’embarquement. Pourquoi on survit plus si on a embarqué a Cherbourg ?

Hypothèse 1: Il y a plus de femmes à Cherbourg (?)

Hypothèse 2: On est plus riche à Cherbourg et plus on est riche plus on a survécu (?)



#### Hypothèse 1: Il y a plus de femmes à Cherbourg

In [ ]:
sex_count = df.groupby(['Embarked', 'Sex'])['PassengerId'].count().unstack(fill_value=0)


sex_count['Total'] = sex_count['female'] + sex_count['male']
sex_count['% Female'] = (sex_count['female'] / sex_count['Total']).round(6)

print(sex_count)


Sex       female  male  Total  % Female
Embarked                               
C            113   157    270  0.418519
Q             60    63    123  0.487805
S            291   623    914  0.318381


Il n’y a pas plus d’individus féminins à Cherbourg qu’à Queenstown.

#### Hypothèse 2: on est plus riche à Cherbourg et plus on est riche plus on a survécu (?)

In [ ]:
summary_pclass = df.groupby('Pclass').agg(
    Survivants=('Survived', 'sum'),
    Passagers=('PassengerId', 'count')
)
summary_pclass['%'] = (summary_pclass['Survivants'] / summary_pclass['Passagers'] * 100).round(1)

print(summary_pclass)


        Survivants  Passagers     %
Pclass                             
1            136.0        323  42.1
2             87.0        277  31.4
3            119.0        709  16.8
